lib/http/intercepted_client.dart

```diff
@@ -264,7 +270,11 @@ class InterceptedClient extends BaseClient {
 
       var stream = requestTimeout == null
           ? await _inner.send(interceptedRequest)
-          : await _inner.send(interceptedRequest).timeout(requestTimeout!);
+          : await _inner.send(interceptedRequest).timeout(requestTimeout!,
+              onTimeout: () async {
```

Comment:
Not entirely sold on this. I'm thinking that maybe we should not set `onTimeout` if our `onRequestTimeout` is null. Is there a specific reason for your implementation or do you think is something we could adjust?